### PR TITLE
Fix/dev 4343 handle no api data

### PR DESCRIPTION
**Pull request by brett-varney**
**High level description:**

Testing revealed error condition not reacting properly, this resolves that

**Technical details:**

adds local state & checks for no data condition

To test, change api request locally to:
```
export const fetchObligationsByAwardType = (code, fy) => {
    return {
        promise: Promise.resolve({
            data: {}
        })
    };
}
```

**JIRA Ticket:**
[DEV-4343](https://federal-spending-transparency.atlassian.net/browse/DEV-4343)

**Mockup:**
in ticket, no change

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [pre-done] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [na] Verified mobile/tablet/desktop/monitor responsiveness
- [na] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [na] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [na] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [na] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [na] Design review complete `if applicable`
- [na] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete

### CODE REVIEW

File: src/js/containers/agencyV2/visualizations/ObligationsByAwardTypeContainer.jsx [hunk 1]
```diff
@@ -43,79 +44,85 @@ export default function ObligationsByAwardTypeContainer({ fiscalYear, windowWidt
...
-                        setError(true);
-                }
-            });
-            setCategoriesForGraph(categories);
-            setDetailsForGraph(details);
-            setLoading(false);
-            obligationsByAwardTypeRequest.current = null;
+            if (Object.keys(res.data).length === 0) {
+                setNoData(true);
+                setLoading(false);
```
Comment (jonhill13) [inline src/js/containers/agencyV2/visualizations/ObligationsByAwardTypeContainer.jsx:49]: Can we add `obligationsByAwardTypeRequest.current = null;` below setLoading

File: src/js/containers/agencyV2/visualizations/ObligationsByAwardTypeContainer.jsx [hunk 2]
```diff
@@ -43,79 +44,85 @@ export default function ObligationsByAwardTypeContainer({ fiscalYear, windowWidt
         };
```
Comment (jonhill13) [inline src/js/containers/agencyV2/visualizations/ObligationsByAwardTypeContainer.jsx:44]: Can we add a conditional here to update noData to false.

**Review by jonhill13 — CHANGES_REQUESTED** [review]

File: src/js/containers/agencyV2/visualizations/ObligationsByAwardTypeContainer.jsx (hunk 1 above)
Comment (brett-varney) [reply src/js/containers/agencyV2/visualizations/ObligationsByAwardTypeContainer.jsx:52]: done

File: src/js/containers/agencyV2/visualizations/ObligationsByAwardTypeContainer.jsx (hunk 2 above)
Comment (brett-varney) [reply src/js/containers/agencyV2/visualizations/ObligationsByAwardTypeContainer.jsx:47]: done

**Review by jonhill13 — DISMISSED** [review]

**Review by jonhill13 — APPROVED** [review]

**Pull request merged**